From 3d7f64bd537f4d5f171835869387c5910c19eb7a Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Mon, 3 Jun 2019 23:00:57 +0800
Subject: [PATCH 2/2] dmaengine: imx-sdma: enable CONFIG_FW_LOADER_USER_HELPER
 for sdma

Enable CONFIG_FW_LOADER_USER_HELPER for sdma since the firmware is
located in rootfs and will be loaded by userspace script through
sysfs interface instead of uevent.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
---
 drivers/dma/Kconfig    | 2 ++
 drivers/dma/imx-sdma.c | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/Kconfig b/drivers/dma/Kconfig
index e8839e8..e45fd93 100644
--- a/drivers/dma/Kconfig
+++ b/drivers/dma/Kconfig
@@ -263,6 +263,8 @@ config IMX_SDMA
 	depends on ARCH_MXC
 	select DMA_ENGINE
 	select DMA_VIRTUAL_CHANNELS
+	select FW_LOADER
+	select FW_LOADER_USER_HELPER
 	help
 	  Support the i.MX SDMA engine. This engine is integrated into
 	  Freescale i.MX25/31/35/51/53/6 chips.
diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 2e05afa..cc08de0 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2128,7 +2128,7 @@ static int sdma_get_firmware(struct sdma_engine *sdma,
 	int ret;
 
 	ret = request_firmware_nowait(THIS_MODULE,
-			FW_ACTION_HOTPLUG, fw_name, sdma->dev,
+			FW_ACTION_NOHOTPLUG, fw_name, sdma->dev,
 			GFP_KERNEL, sdma, sdma_load_firmware);
 
 	return ret;
-- 
2.7.4

